### DISCUSSION TITLE
Install error

### DISCUSSION THREAD

**Discussion by qness** [RhetTbull/osxphotos#819] [Q&A] [answered]
Hi, 

I'm trying to install osxphotos using the "If you aren't familiar with installing python applications" instructions.  I successfully installed Homebrew and pipx but get the following output when running pipx install osxphotos...

~~~~~~~~~~~~~~~~~~~~
Fatal error from pip prevented installation. Full pip output in file:
    /Users/dave/.local/pipx/logs/cmd_2022-11-07_00.39.19_pip_errors.log

pip seemed to fail to build package:
    pyobjc-framework-Cocoa<9.0,>=7.3

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    TypeError: not enough arguments for format string
    error: metadata-generation-failed

Error installing osxphotos.
~~~~~~~~~~~~~~~~~~~~

Log files attached.  

Any help is appreciated.

[cmd_2022-11-07_00.39.19_pip_errors.log](https://github.com/RhetTbull/osxphotos/files/9949693/cmd_2022-11-07_00.39.19_pip_errors.log)
[cmd_2022-11-07_00.39.19.log](https://github.com/RhetTbull/osxphotos/files/9949694/cmd_2022-11-07_00.39.19.log)

**Comment by RhetTbull** [comment]
You're using python 3.11 and some of the osxphotos dependencies do not work with 3.11.  Please install 3.10 and try again.  

Did you install python 3.11 or did brew do that when install pipx?  

I've not tested it but this might work:

`brew uninstall pipx`
`brew uninstall python@3.11`
`brew install python@3.10`
`brew install pipx`

**Reply by pylchung** [reply]
I am getting the exact same error. Python 3.11 is being installed by pipx. MacOS version is Ventura on an Mac Mini M1.

**Comment by pylchung** [comment, marked as answer]
I got around this issue by using the following

`pipx install --python python3.10 osxphotos`

If Python 3.10 is not installed already, run

`brew install python@3.10`

**Reply by RhetTbull** [reply]
Thanks! I'll update the README

**Comment by qness** [comment]
I got it installed, but it wasn't pretty.  @pylchung's solution is much better than what I did, but I'll recount it in case there is any worthwhile information there.  

The original sequence of commands suggested by @RhetTbull did not work because after uninstalling python 3.11 and installing 3.10, the "brew install pipx" command caused python to be upgraded to 3.11.  After I uninstalled pipx and python 3.11 again, the following sequence worked.

brew reinstall python@3.10
pip3 install osxphotos

I then tried to upgrade to the latest version of osxphotos. 
_**pipx upgrade osxphotos**_  failed with zsh: command not found: pipx

_**pip3 upgrade osxphotos**_  failed ERROR: unknown command "upgrade"

_**osxphotos -v**_  returned osxphotos, version 0.52.0.  

Is this the latest?  If not, or for future reference, how do I upgrade to the latest version?

Otherwise, all good.  I'm enjoying the app.

**Reply by RhetTbull** [reply]
Glad you got it sorted out. Yes, 0.52.0 is the latest version.
